### ISSUE TITLE
Segfault with kops create -f ...

### ISSUE THREAD

**Issue by Gottox** [kubernetes/kops#5926]
**1. What `kops` version are you running? The command `kops version`, will display
 this information.**

Version 1.10.0 (git-8b52ea6d1)

**2. What Kubernetes version are you running? `kubectl version` will print the
 version if a cluster is running or provide the Kubernetes version specified as
 a `kops` flag.**

```
kubectl version --client=true
Client Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.1", GitCommit:"4ed3216f3ec431b140b1d899130a69fc671678f4", GitTreeState:"archive", BuildDate:"2018-10-07T09:55:32Z", GoVersion:"go1.11", Compiler:"gc", Platform:"linux/amd64"}
```

**3. What cloud provider are you using?**

AWS

**4. What commands did you run?  What is the simplest way to reproduce this issue?**

```
$ cat > config <<EOF
apiVersion: kops/v1alpha2
kind: Cluster
metadata:
  name: cluster.foo.bar
spec:
  api:
    loadBalancer:
      type: Internal
  autorization:
    rbac: {}
  channel: stable
  cloudProvider: aws
  configBase: s3://clusters.foo.bar/cluster.foo.bar
  iam:
    allowContainerRegistry: true
    legacy: false
  kubernetesApiAccess:
  - 0.0.0.0/0
  topology:
    dns:
      type: Public
    masters: private
    nodes: private
  etcdClusters:
  - etcdMembers:
    - instanceGroup: master-eu-central-1a
      name: a
    - instanceGroup: master-eu-central-1b
      name: b
    - instanceGroup: master-eu-central-1c
      name: c
    name: main
  - etcdMembers:
    - instanceGroup: master-eu-central-1a
      name: a
    - instanceGroup: master-eu-central-1b
      name: b
    - instanceGroup: master-eu-central-1c
      name: c
    name: events
  subnets:
  - cidr: 172.20.32.0/19
    name: eu-central-1a
    type: Private
    zone: eu-central-1a
  - cidr: 172.20.64.0/19
    name: eu-central-1b
    type: Private
    zone: eu-central-1b
  - cidr: 172.20.96.0/19
    name: eu-central-1c
    type: Private
    zone: eu-central-1c
  - cidr: 172.20.0.0/22
    name: utility-eu-central-1a
    type: Utility
    zone: eu-central-1a
  - cidr: 172.20.4.0/22
    name: utility-eu-central-1b
    type: Utility
    zone: eu-central-1b
  - cidr: 172.20.8.0/22
    name: utility-eu-central-1c
    type: Utility
    zone: eu-central-1c

---

apiVersion: kops/v1alpha2
kind: InstanceGroup
metadata:
  labels:
    kops.k8s.io/cluster: cluster.foo.bar
  name: bastions
spec:
  image: kope.io/k8s-1.9-debian-jessie-amd64-hvm-ebs-2018-03-11
  machineType: t2.micro
  maxSize: 1
  minSize: 1
  nodeLabels:
    kops.k8s.io/instancegroup: bastions
  role: Bastion
  subnets:
  - eu-central-1a
  - eu-central-1b
  - eu-central-1c

---

apiVersion: kops/v1alpha2
kind: InstanceGroup
metadata:
  labels:
    kops.k8s.io/cluster: cluster.foo.bar
  name: master-eu-central-1a
spec:
  image: kope.io/k8s-1.9-debian-jessie-amd64-hvm-ebs-2018-03-11
  machineType: m3.medium
  maxSize: 1
  minSize: 1
  nodeLabels:
    kops.k8s.io/instancegroup: master-eu-central-1a
  role: Master
  subnets:
  - eu-central-1a

---

apiVersion: kops/v1alpha2
kind: InstanceGroup
metadata:
  labels:
    kops.k8s.io/cluster: cluster.foo.bar
  name: master-eu-central-1b
spec:
  image: kope.io/k8s-1.9-debian-jessie-amd64-hvm-ebs-2018-03-11
  machineType: m3.medium
  maxSize: 1
  minSize: 1
  nodeLabels:
    kops.k8s.io/instancegroup: master-eu-central-1b
  role: Master
  subnets:
  - eu-central-1b

---

apiVersion: kops/v1alpha2
kind: InstanceGroup
metadata:
  labels:
    kops.k8s.io/cluster: cluster.foo.bar
  name: master-eu-central-1c
spec:
  image: kope.io/k8s-1.9-debian-jessie-amd64-hvm-ebs-2018-03-11
  machineType: m3.medium
  maxSize: 1
  minSize: 1
  nodeLabels:
    kops.k8s.io/instancegroup: master-eu-central-1c
  role: Master
  subnets:
  - eu-central-1c

---

apiVersion: kops/v1alpha2
kind: InstanceGroup
metadata:
  labels:
    kops.k8s.io/cluster: cluster.foo.bar
  name: nodes-eu-central-1a
spec:
  image: kope.io/k8s-1.9-debian-jessie-amd64-hvm-ebs-2018-03-11
  machineType: m5.xlarge
  maxSize: 5
  minSize: 2
  nodeLabels:
    kops.k8s.io/instancegroup: nodes-eu-central-1a
  role: Node
  subnets:
  - eu-central-1a

---

apiVersion: kops/v1alpha2
kind: InstanceGroup
metadata:
  labels:
    kops.k8s.io/cluster: cluster.foo.bar
  name: nodes-eu-central-1b
spec:
  image: kope.io/k8s-1.9-debian-jessie-amd64-hvm-ebs-2018-03-11
  machineType: m5.xlarge
  maxSize: 5
  minSize: 2
  nodeLabels:
    kops.k8s.io/instancegroup: nodes-eu-central-1b
  role: Node
  subnets:
  - eu-central-1b

---

apiVersion: kops/v1alpha2
kind: InstanceGroup
metadata:
  labels:
    kops.k8s.io/cluster: cluster.foo.bar
  name: nodes-eu-central-1c
spec:
  image: kope.io/k8s-1.9-debian-jessie-amd64-hvm-ebs-2018-03-11
  machineType: m5.xlarge
  maxSize: 5
  minSize: 2
  nodeLabels:
    kops.k8s.io/instancegroup: nodes-eu-central-1c
  role: Node
  subnets:
  - eu-central-1c
EOF
$ kops create -f config
```

**5. What happened after the commands executed?**

The following error message appears:
```
$ kops create -f f
I1012 14:11:16.471841   14972 defaults.go:121] Using KubernetesVersion "1.10.6" from channel "stable"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x1ddefca]

goroutine 1 [running]:
k8s.io/kops/pkg/apis/kops/validation.ValidateCluster(0xc420223000, 0xc42060a300, 0xc420965750)
	/go/src/k8s.io/kops/pkg/apis/kops/validation/legacy.go:236 +0x8aa
k8s.io/kops/pkg/client/simple/vfsclientset.(*ClusterVFS).Create(0xc4209827d0, 0xc420223000, 0xc4202231a8, 0x1, 0xc420c78e70)
	/go/src/k8s.io/kops/pkg/client/simple/vfsclientset/cluster.go:101 +0x60
k8s.io/kops/pkg/client/simple/vfsclientset.(*VFSClientset).CreateCluster(0xc420c766e0, 0xc420223000, 0x0, 0x1000, 0xc420c78e70)
	/go/src/k8s.io/kops/pkg/client/simple/vfsclientset/clientset.go:56 +0x43
main.RunCreate(0xc420c760a0, 0x538b6a0, 0xc42000c018, 0xc420c761e0, 0x0, 0x0)
	/go/src/k8s.io/kops/cmd/kops/create.go:165 +0x446
main.NewCmdCreate.func1(0xc42010cf00, 0xc420c76680, 0x0, 0x2)
	/go/src/k8s.io/kops/cmd/kops/create.go:95 +0x66
k8s.io/kops/vendor/github.com/spf13/cobra.(*Command).execute(0xc42010cf00, 0xc420c76620, 0x2, 0x2, 0xc42010cf00, 0xc420c76620)
	/go/src/k8s.io/kops/vendor/github.com/spf13/cobra/command.go:757 +0x2c1
k8s.io/kops/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x55dc440, 0x2d38700, 0x0, 0x0)
	/go/src/k8s.io/kops/vendor/github.com/spf13/cobra/command.go:843 +0x334
k8s.io/kops/vendor/github.com/spf13/cobra.(*Command).Execute(0x55dc440, 0x5612040, 0x0)
	/go/src/k8s.io/kops/vendor/github.com/spf13/cobra/command.go:791 +0x2b
main.Execute()
	/go/src/k8s.io/kops/cmd/kops/root.go:96 +0x91
main.main()
	/go/src/k8s.io/kops/cmd/kops/main.go:25 +0x20
```

**6. What did you expect to happen?**

kops should start to configure the cluster

**7. Please provide your cluster manifest. Execute
  `kops get --name my.example.com -o yaml` to display your cluster manifest.
  You may want to remove your cluster name and other sensitive information.**

No cluster exists yet.

**8. Please run the commands with most verbose logging by adding the `-v 10` flag.
  Paste the logs into this report, or in a gist and provide the gist link here.**

```
$ kops create -f config -v 10
I1012 14:12:52.342665   15048 factory.go:68] state store s3://clusters.infrastructure.vwn.cloud
I1012 14:12:52.356411   15048 aws_cloud.go:984] Querying EC2 for all valid zones in region "eu-central-1"
I1012 14:12:52.356511   15048 request_logger.go:45] AWS request: ec2/DescribeAvailabilityZones
I1012 14:12:52.849244   15048 subnets.go:49] All subnets have CIDRs; skipping assignment logic
I1012 14:12:52.849326   15048 defaults.go:213] Not setting up Proxy Excludes
I1012 14:12:52.849383   15048 channel.go:97] resolving "stable" against default channel location "https://raw.githubusercontent.com/kubernetes/kops/master/channels/"
I1012 14:12:52.849460   15048 channel.go:102] Loading channel from "https://raw.githubusercontent.com/kubernetes/kops/master/channels/stable"
I1012 14:12:52.849506   15048 context.go:159] Performing HTTP request: GET https://raw.githubusercontent.com/kubernetes/kops/master/channels/stable
I1012 14:12:53.406818   15048 channel.go:111] Channel contents: spec:
  images:
    # We put the "legacy" version first, for kops versions that don't support versions ( < 1.5.0 )
    - name: kope.io/k8s-1.4-debian-jessie-amd64-hvm-ebs-2017-07-28
      providerID: aws
      kubernetesVersion: ">=1.4.0 <1.5.0"
    - name: kope.io/k8s-1.5-debian-jessie-amd64-hvm-ebs-2018-03-11
      providerID: aws
      kubernetesVersion: ">=1.5.0 <1.6.0"
    - name: kope.io/k8s-1.6-debian-jessie-amd64-hvm-ebs-2018-03-11
      providerID: aws
      kubernetesVersion: ">=1.6.0 <1.7.0"
    - name: kope.io/k8s-1.7-debian-jessie-amd64-hvm-ebs-2018-03-11
      providerID: aws
      kubernetesVersion: ">=1.7.0 <1.8.0"
    - name: kope.io/k8s-1.8-debian-jessie-amd64-hvm-ebs-2018-03-11
      providerID: aws
      kubernetesVersion: ">=1.8.0 <1.9.0"
    - name: kope.io/k8s-1.9-debian-jessie-amd64-hvm-ebs-2018-03-11
      providerID: aws
      kubernetesVersion: ">=1.9.0 <1.10.0"
    # Need stretch as default in 1.10 (for nvme)
    # BUT... this is causing the submit queue to block, so back to jessie temporarily: https://github.com/kubernetes/kubernetes/issues/56763
    - name: kope.io/k8s-1.9-debian-jessie-amd64-hvm-ebs-2018-03-11
      providerID: aws
      kubernetesVersion: ">=1.10.0"
    - providerID: gce
      name: "cos-cloud/cos-stable-60-9592-90-0"
  cluster:
    kubernetesVersion: v1.5.8
    networking:
      kubenet: {}
  kubernetesVersions:
  - range: ">=1.11.0"
    recommendedVersion: 1.11.2
    requiredVersion: 1.11.0
  - range: ">=1.10.0"
    recommendedVersion: 1.10.6
    requiredVersion: 1.10.0
  - range: ">=1.9.0"
    recommendedVersion: 1.9.10
    requiredVersion: 1.9.0
  - range: ">=1.8.0"
    recommendedVersion: 1.8.15
    requiredVersion: 1.8.0
  - range: ">=1.7.0"
    recommendedVersion: 1.7.16
    requiredVersion: 1.7.0
  - range: ">=1.6.0"
    recommendedVersion: 1.6.13
    requiredVersion: 1.6.0
  - range: ">=1.5.0"
    recommendedVersion: 1.5.8
    requiredVersion: 1.5.1
  - range: "<1.5.0"
    recommendedVersion: 1.4.12
    requiredVersion: 1.4.2
  kopsVersions:
  - range: ">=1.11.0-alpha.1"
    #recommendedVersion: "1.10.0"
    #requiredVersion: 1.10.0
    kubernetesVersion: 1.11.2
  - range: ">=1.10.0-alpha.1"
    recommendedVersion: "1.10.0"
    #requiredVersion: 1.10.0
    kubernetesVersion: 1.10.6
  - range: ">=1.9.0-alpha.1"
    recommendedVersion: 1.9.2
    #requiredVersion: 1.9.0
    kubernetesVersion: 1.9.10
  - range: ">=1.8.0-alpha.1"
    recommendedVersion: 1.8.1
    requiredVersion: 1.7.1
    kubernetesVersion: 1.8.15
  - range: ">=1.7.0-alpha.1"
    recommendedVersion: 1.8.1
    requiredVersion: 1.7.1
    kubernetesVersion: 1.7.16
  - range: ">=1.6.0-alpha.1"
    recommendedVersion: 1.8.1
    requiredVersion: 1.7.1
    kubernetesVersion: 1.6.13
  - range: ">=1.5.0-alpha1"
    recommendedVersion: 1.8.1
    requiredVersion: 1.7.1
    kubernetesVersion: 1.5.8
  - range: "<1.5.0"
    recommendedVersion: 1.8.1
    requiredVersion: 1.7.1
    kubernetesVersion: 1.4.12
I1012 14:12:53.406920   15048 channel.go:239] version range ">=1.11.0-alpha.1" does not apply to version "1.10.0"; skipping
I1012 14:12:53.406941   15048 defaults.go:121] Using KubernetesVersion "1.10.6" from channel "stable"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x1ddefca]

goroutine 1 [running]:
k8s.io/kops/pkg/apis/kops/validation.ValidateCluster(0xc42060f000, 0xc420ccb100, 0xc42094b750)
	/go/src/k8s.io/kops/pkg/apis/kops/validation/legacy.go:236 +0x8aa
k8s.io/kops/pkg/client/simple/vfsclientset.(*ClusterVFS).Create(0xc420cb6870, 0xc42060f000, 0xc42060f1a8, 0x1, 0xc420c4fc50)
	/go/src/k8s.io/kops/pkg/client/simple/vfsclientset/cluster.go:101 +0x60
k8s.io/kops/pkg/client/simple/vfsclientset.(*VFSClientset).CreateCluster(0xc420c616a0, 0xc42060f000, 0x0, 0x1000, 0xc420c4fc50)
	/go/src/k8s.io/kops/pkg/client/simple/vfsclientset/clientset.go:56 +0x43
main.RunCreate(0xc420c610a0, 0x538b6a0, 0xc42008a008, 0xc420c611e0, 0x0, 0x0)
	/go/src/k8s.io/kops/cmd/kops/create.go:165 +0x446
main.NewCmdCreate.func1(0xc420376000, 0xc420c8eb40, 0x0, 0x4)
	/go/src/k8s.io/kops/cmd/kops/create.go:95 +0x66
k8s.io/kops/vendor/github.com/spf13/cobra.(*Command).execute(0xc420376000, 0xc420c8ea80, 0x4, 0x4, 0xc420376000, 0xc420c8ea80)
	/go/src/k8s.io/kops/vendor/github.com/spf13/cobra/command.go:757 +0x2c1
k8s.io/kops/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x55dc440, 0x2d38700, 0x0, 0x0)
	/go/src/k8s.io/kops/vendor/github.com/spf13/cobra/command.go:843 +0x334
k8s.io/kops/vendor/github.com/spf13/cobra.(*Command).Execute(0x55dc440, 0x5612040, 0x0)
	/go/src/k8s.io/kops/vendor/github.com/spf13/cobra/command.go:791 +0x2b
main.Execute()
	/go/src/k8s.io/kops/cmd/kops/root.go:96 +0x91
main.main()
	/go/src/k8s.io/kops/cmd/kops/main.go:25 +0x20
```

**9. Anything else do we need to know?**

/

**Comment by gambol99** [comment]
Looks like it was fixed in https://github.com/kubernetes/kops/pull/5801 but no tagged release. An fix workaround would be to just select a [networking cni](https://github.com/kubernetes/kops/blob/master/docs/networking.md)

**Comment by Gottox** [comment]
resolved in #5801
